scripts/generate-javadoc.sh

```diff
@@ -16,7 +16,12 @@ mkdir_p "$ARCHIVE_DIR"
 function generate_javadoc_core() {
 	declare release=$1
 	# First we need to get the built javadocs.
-	wget --no-verbose "https://repo.jenkins-ci.org/releases/org/jenkins-ci/main/jenkins-core/${release}/jenkins-core-${release}-javadoc.jar"
+	if [ -n ${ARTIFACT_CACHING_PROXY_ORIGIN-} ]
```

Comment:
I don't understand the trailing `-` character.  When I try that in my shell, it seems to always evaluate to true, whether the variable is set or not
```suggestion
	if [ -n ${ARTIFACT_CACHING_PROXY_ORIGIN-} ]
```

Reply:
@MarkEWaite I've learned about this trailing dash from this PR: https://github.com/jenkinsci/bom/pull/1149#issue-1241966546
> The use of the dash is explained in the "Rewriting scripts to deal with it" section of [Bash FAQ #112](https://mywiki.wooledge.org/BashFAQ/112).

I don't see any change in your suggestion, should there be one?

Reply:
@MarkEWaite you were right, there was an error, explained and fixed here: https://github.com/jenkins-infra/javadoc/pull/59#discussion_r1139996715